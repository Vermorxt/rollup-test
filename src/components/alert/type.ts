import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
