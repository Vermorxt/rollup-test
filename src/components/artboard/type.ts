import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
