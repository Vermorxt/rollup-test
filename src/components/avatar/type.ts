import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
