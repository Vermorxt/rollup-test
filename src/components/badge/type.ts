import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
