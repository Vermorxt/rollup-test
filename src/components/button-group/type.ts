import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
