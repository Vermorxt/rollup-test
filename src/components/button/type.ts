import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
