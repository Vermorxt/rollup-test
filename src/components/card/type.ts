import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
