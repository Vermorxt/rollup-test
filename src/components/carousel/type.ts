import { CSSProperties, HTMLProps } from 'react'
export interface Ui_CarouselProps extends HTMLProps<HTMLDivElement> {
  center?: boolean
  vertical?: boolean
  end?: boolean
  full?: boolean
  fullHeight?: boolean
  halfWidth?: boolean
  rounded?: boolean
  relative?: boolean
}
export interface Ui_CarouselArrowNavigationProps {
  index: number
  itemLength: number
  usedId: string
  style?: CSSProperties
  className?: string
}
