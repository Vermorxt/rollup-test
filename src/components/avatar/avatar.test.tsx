import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { checkAccessibility, itSupportsSystemProps } from '../../_tests-utils'
import { itHasCssClass } from '../../_tests-utils/it-has-color-css-class'
import { Ui_AvatarProps } from './type'
import Ui_Avatar from '../avatar'
import React from 'react'

const defaultProps: Ui_AvatarProps = {
  roundedXl: false,
  rounded2xl: false,
  rounded3xl: false,
  roundedLg: false,
  roundedFull: false,
  ringOffset: '8',
}

const selector = 'avatar'

describe('@cms/core/Ui_Avatar', () => {
  checkAccessibility([<Ui_Avatar {...defaultProps} ref={null} />])
  itSupportsSystemProps({
    component: Ui_Avatar,
    props: defaultProps,
    refType: HTMLDivElement,
  })

  // NOTE: (possible debug): debugComponent(<Ui_Avatar {...defaultProps} ref={null} />)

  it('should match snapshot', () => {
    const { container } = render(<Ui_Avatar {...defaultProps} horizontal ref={null} />)
    expect(container).toMatchSnapshot()
  })

  it('renders ONLINE status based on prop', () => {
    const { container: available } = render(<Ui_Avatar {...defaultProps} online ref={null} />)
    const { container: notAvailable } = render(<Ui_Avatar {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.online`)).toHaveLength(1)
    expect(notAvailable.querySelector(`.online`)).toBeNull()
  })

  it('renders OFFLINE status based on prop', () => {
    const { container: available } = render(<Ui_Avatar {...defaultProps} offline ref={null} />)
    const { container: notAvailable } = render(<Ui_Avatar {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.offline`)).toHaveLength(1)
    expect(notAvailable.querySelector(`.offline`)).toBeNull()
  })

  it('renders usePlaceHolder status based on prop', () => {
    const { container: available } = render(<Ui_Avatar {...defaultProps} usePlaceHolder ref={null} />)
    const { container: notAvailable } = render(<Ui_Avatar {...defaultProps} ref={null} />)

    expect(available.querySelectorAll(`.bg-neutral-focus.text-neutral-content`)).toHaveLength(1)
    expect(notAvailable.querySelector(`.bg-neutral-focus.text-neutral-content`)).toBeNull()
  })

  it('renders SIZE classes', async () => {
    await itHasCssClass(<Ui_Avatar large />, '.w-32')
    await itHasCssClass(<Ui_Avatar medium />, '.w-20')
    await itHasCssClass(<Ui_Avatar small />, '.w-16')
    await itHasCssClass(<Ui_Avatar mini />, '.w-12')
    await itHasCssClass(<Ui_Avatar tiny />, '.w-8')
  })

  it('renders SHAPE classes', async () => {
    await itHasCssClass(<Ui_Avatar triangle />, '.mask-triangle')
    await itHasCssClass(<Ui_Avatar squircle />, '.mask-squircle')
    await itHasCssClass(<Ui_Avatar hexagon />, '.mask-hexagon')
  })

  it('renders RING classes', async () => {
    const offset = '8'
    const color = 'primary'

    await itHasCssClass(<Ui_Avatar ring />, '.ring')
    await itHasCssClass(<Ui_Avatar ring ringPrimary />, '.ring-primary')
    await itHasCssClass(<Ui_Avatar ring ringSecondary />, '.ring-secondary')
    await itHasCssClass(<Ui_Avatar ring ringOffset={offset} />, `.ring-offset-${offset}`)
    await itHasCssClass(<Ui_Avatar ring ringOffset={offset} ringOffsetBackground={color} />, `.ring-offset-${color}`)
  })

  it('renders BORDER_RADIUS classes', async () => {
    await itHasCssClass(<Ui_Avatar roundedXl />, `.rounded-xl`)
    await itHasCssClass(<Ui_Avatar rounded2xl />, `.rounded-2xl`)
    await itHasCssClass(<Ui_Avatar rounded3xl />, `.rounded-3xl`)
    await itHasCssClass(<Ui_Avatar roundedLg />, `.rounded-lg`)
    await itHasCssClass(<Ui_Avatar roundedFull />, `.rounded-full`)
  })
})
