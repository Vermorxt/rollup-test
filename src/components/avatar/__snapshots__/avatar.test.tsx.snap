// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Avatar should match snapshot 1`] = `
<div>
  <div
    class="
        avatar
         
         
         
        "
  >
    <div
      class="
         
         w-24 
        rounded-xl,rounded-2xl,rounded-3xl,rounded-lg,rounded-full  
          
        "
    >
      <img
        alt=""
        src="/static/media/png/placeholder.png"
      />
    </div>
  </div>
</div>
`;
